packages/snaps-execution-environments/webpack.config.js

```diff
@@ -0,0 +1,296 @@
+// Ideally we write this file with TypeScript, but `@lavamoat/webpack` doesn't
+// have any declarations, and it currently makes using it from LavaMoat Node
+// more difficult.
+
+const LavaMoatPlugin = require('@lavamoat/webpack');
+const { readFileSync } = require('fs');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const { isBuiltin } = require('module');
+const { resolve } = require('path');
+const { TsconfigPathsPlugin } = require('tsconfig-paths-webpack-plugin');
+const { ProvidePlugin, Compilation } = require('webpack');
+const { merge } = require('webpack-merge');
+
+/**
+ * Whether to generate a policy file for the build.
+ *
+ * @type {boolean}
+ */
+// eslint-disable-next-line n/no-process-env
+const UPDATE_POLICY = process.env.LAVAMOAT_GENERATE_POLICY === 'true';
+
+/**
+ * Whether the build is for production or development.
+ *
+ * @type {boolean}
+ */
+const IS_PRODUCTION =
+  // eslint-disable-next-line n/no-process-env
+  process.env.NODE_ENV !== 'development' && process.env.NODE_ENV !== 'test';
+
+/**
+ * The SES bundle used for the lockdown script.
+ *
+ * @type {string}
+ */
+// eslint-disable-next-line n/no-sync
+const SES_BUNDLE = readFileSync(require.resolve('ses'), 'utf-8');
+
+/**
+ * @typedef {import('webpack').Configuration} Configuration
+ */
+
+/**
+ * @typedef {object} EntryPoint
+ * @property {string} name - The name of the entry point.
+ * @property {string} entry - The entry point file.
+ * @property {boolean} [inline] - Whether to inline the lockdown script.
+ * @property {boolean} [inlineBundle] - Whether to inline the bundle.
+ * @property {boolean} [scuttleGlobalThis] - Whether to enable scuttling.
+ * @property {string[]} [scuttleGlobalThisExceptions] - Exceptions to scuttling,
+ * in addition to the default ones.
+ * @property {import('webpack').Configuration} [config] - Additional webpack
+ * configuration for this entry point. This is merged with the base
+ * configuration.
+ */
+
+/**
+ * The default Webpack configuration for web-based entry points.
+ *
+ * @type {Configuration}
+ */
+const DEFAULT_WEB_CONFIG = {
+  target: `browserslist:${resolve(__dirname, '.browserslistrc')}`,
+  plugins: [
+    new HtmlWebpackPlugin({
+      template: './src/index.ejs',
+      templateParameters: () => {
+        return {
+          lockdownScript: SES_BUNDLE,
+        };
+      },
+    }),
+  ],
+};
+
+/**
+ * The entry points for the Webpack build.
+ *
+ * @type {EntryPoint[]}
+ */
+const ENTRY_POINTS = [
+  {
+    name: 'iframe',
+    entry: './src/iframe/index.ts',
+    scuttleGlobalThis: true,
```

Comment:
Does omitting the exceptions mean we don't run with any exceptions in the iframe?

Reply:
There are some default exceptions in the base config.

Reply:
Ah, sorry just got to that part 😄